File: lab2_task1/main.js
function checkBoardBounds(n, m, steps) {
    for (let startX = 1; startX <= n; startX++) {
    for (let startY = 1; startY <= m; startY++) {
    let x = startX, y = startY;
    let isValid = true;
    for (let step of steps) {
    switch (step) {
    case 'L': x--; break;
    case 'R': x++; break;
    case 'D': y--; break;
    case 'U': y++; break;
    default: return "Ошибка в ходах";
    }
    if (x < 1 || x > n || y < 1 || y > m) {
    isValid = false;
    break;
    }
    }
    if (isValid) return `(${startX},${startY})`;
    }
    }
    return "No";
    }
    
    const readline = require('readline');
    const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
    });
    
    rl.question('Введите длину доски: ', n => {
    rl.question('Введите ширину доски: ', m => {
    rl.question('Введите ходы для фигуры (LRDU): ', steps => {
    console.log(checkBoardBounds(n, m, steps));
    rl.close();
    });
    });
    });

File: lab2_task1/main.py
def checkBoardBounds (n, m, steps):
    for startX in range(1, n):
        for startY in range(1, m):
            x = startX
            y = startY
            isValid = True
            
            for step in steps:
                if step == 'L': x -= 1
                elif step == 'R': x += 1
                elif step == 'D': y -= 1
                elif step == 'U': y += 1
                if x < 1 or x > n or y < 1 or y > m:
                    isValid = False
                    break
            if isValid:
                return (f'({startX}, {startY})')
    return "No"


n = int(input("Введите длину доски: ")) 
m = int(input("Введите ширину доски: "))
steps = input("Введите ходы для фигуры (LRDU): ")
print(checkBoardBounds(n, m, steps))